### DISCUSSION TITLE
SHA256, MD5 hashes...

### DISCUSSION THREAD

**Discussion by jgratero** [crosswire/xiphos#1079] [Q&A]
Where can I find the SHA256, MD5 hashes for the current Windows releases of Xiphos?

**Comment by alerque** [comment]
We don't currently publicize any hashes. Given the way releases currently happen (fully automated from a Git tag, generated remotely and automatically posted as build artifacts) there wouldn't really be a whole lot of security gained by posting hashes. One of the things they are useful for is making sure remote sources haven't been tampered with, but to really be effective they would need to be computed by a different system and posted on a different channel. Or for that purpose, signed by a developer. If there is a serious need to ward off some specific threat model feel free to post an issue and we'll figure out how to address it.

The other thing they do is confirm you don't have a broken download. The package wrappers tend to do that already, but you can also confirm by downloading and hashing yourself on a separate device.

**Comment by greg-hellings** [comment]
That said, since the releases are automatic, adding hash values to the
release message in Github shouldn't be difficult, either.

--Greg

**Reply by alerque** [reply]
Hashes would be easy but not add a lot of value. Signatures are harder to do *right* since they pretty much have to be done on a local machine to mean much. I do that for a number of projects where I download and verify otherwise automated releases then post back a signature others can use if they want. That kind of pattern could be used here.
